controllers/keycloakclient/chain/process_scope_test.go

```diff
@@ -0,0 +1,136 @@
+package chain
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Nerzal/gocloak/v12"
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	keycloakApi "github.com/epam/edp-keycloak-operator/api/v1"
+	"github.com/epam/edp-keycloak-operator/pkg/client/keycloak"
+	"github.com/epam/edp-keycloak-operator/pkg/client/keycloak/mocks"
+)
+
+func TestProcessScope_Serve(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name              string
+		keycloakClient    *keycloakApi.KeycloakClient
+		keycloakApiClient func(t *testing.T) keycloak.Client
+		wantErr           require.ErrorAssertionFunc
+	}{
+		{
+			name:           "client has no authorization settings",
+			keycloakClient: &keycloakApi.KeycloakClient{},
+			keycloakApiClient: func(t *testing.T) keycloak.Client {
+				return mocks.NewMockClient(t)
+			},
+			wantErr: require.NoError,
+		},
+		{
+			name: "get scopes",
+			keycloakClient: &keycloakApi.KeycloakClient{
+				Spec: keycloakApi.KeycloakClientSpec{
+					ClientId: "client",
+					Authorization: &keycloakApi.Authorization{
+						Scopes: []string{"scope-auth1"},
+					},
+				},
+			},
+			keycloakApiClient: func(t *testing.T) keycloak.Client {
+				client := mocks.NewMockClient(t)
+				client.On("GetClientID", "client", "master").
+					Return("clientID", nil).Once()
+				client.On("GetScopes", mock.Anything, "master", "clientID").
+					Return(map[string]gocloak.ScopeRepresentation{
+						"scope-auth": {
+							Name: gocloak.StringP("scope-auth1"),
```

Comment:
Add `ID` property.
Without ID you have panic here
`if err := h.keycloakApiClient.DeleteScope(ctx, realmName, clientID, *existingScopes[name].ID)`